Mathlib/Computability/DFA.lean

```diff
@@ -91,12 +92,22 @@ theorem evalFrom_of_append (start : σ) (x y : List α) :
   x.foldl_append _ _ y
 #align DFA.eval_from_of_append DFA.evalFrom_of_append
 
+/--
+`M.acceptsFrom start` is the language of `x` such that `M.evalFrom start x` is an accept state.
+-/
+def acceptsFrom (start : σ) : Language α := {x | M.evalFrom start x ∈ M.accept}
+
+theorem mem_acceptsFrom (start : σ) (x : List α) :
+    x ∈ M.acceptsFrom start ↔ M.evalFrom start x ∈ M.accept := by rfl
+
 /-- `M.accepts` is the language of `x` such that `M.eval x` is an accept state. -/
-def accepts : Language α := {x | M.eval x ∈ M.accept}
+def accepts : Language α := M.acceptsFrom M.start
 #align DFA.accepts DFA.accepts
 
-theorem mem_accepts (x : List α) : x ∈ M.accepts ↔ M.evalFrom M.start x ∈ M.accept := by rfl
-#align DFA.mem_accepts DFA.mem_accepts
+theorem mem_accepts (x : List α) : x ∈ M.accepts ↔ M.eval x ∈ M.accept := by rfl
+
+theorem mem_accepts' (x : List α) : x ∈ M.accepts ↔ M.evalFrom M.start x ∈ M.accept := by rfl
+#align DFA.mem_accepts DFA.mem_accepts'
```

Comment:
I don't think you need `mem_accepts'`
```suggestion
theorem mem_accepts (x : List α) : x ∈ M.accepts ↔ M.eval x ∈ M.accept := by rfl
#align DFA.mem_accepts DFA.mem_accepts
```

Reply:
Sounds good. It was used in `pumping_lemma` but I replaced it with `rw [mem_accepts, eval]`.